pep-0611.rst

```diff
@@ -190,32 +206,49 @@ It is anticipated that CPython will enforce the limits as follows:
 * The sum of local variables and stack usage for a code object: 3.9 onward.
 * The number of classes in a running interpreter: probably 3.10 onward, maybe warning in 3.9.
 
-Backwards Compatibility
-=======================
+Hard limits in CPython
+======================
+
+CPython will enforce a hard limit on all the above values. The value of the hard limit will be 8 million.
 
 It is hypothetically possible that some machine generated code exceeds one or more of the above limits.
-The author believes that to be highly unlikely and easily fixed by modifying the output stage of the code generator.
+The author believes that to be incredibly unlikely and easily fixed by modifying the output stage of the code generator.
 
 We would like to gain the benefit from the above limits for performance as soon as possible.
 To that end, CPython will start applying limits from version 3.9 onward.
-To ease the transition and minimize breakage, the initial limits would be much larger than one million
-and will be gradually reduced to one million over several versions.
+To ease the transition and minimize breakage, the initial limits will be 16 million, reducing to 8 million in a later version.
 
-The actual limits enforced by CPython will be:
+Backwards Compatibility
+=======================
+
+The actual hard limits enforced by CPython will be:
 
 =============  ===============
-   Version      Limit
+   Version      Hard limit
 =============  ===============
- 3.9           8 million
- 3.10          4 million
- 3.11          2 million
- 3.12 onward   1 million
+ 3.9            16 million
+ 3.10 onward    8 million
 =============  ===============
 
 Given the rarity of code generators that would exceed the one million limits,
 and the environments in which they are typically used, it seems reasonable
 to start issuing warnings in 3.9 if any limited quantity exceeds one million.
 
+Historically the recursion limit has been set at 1000. To avoid breaking code that implicitly relies on the value being small,
+the soft recursion limit will be increased gradually, as follows:
```

Comment:
How do you plan to implement this?

Reply:
The Python call stack resides on the heap. Changing the interpreter to handle Python calls without consuming C stack is a reasonably straightforward transformation.

As an aside, separating the protection from runaway recursion and C-stack overflow protection would be a good idea regardless.

Reply:
I would be happy to support this *after* a working implementation for CPython has been provided as a PR. Until then I am skeptical. In particular, the C API concerns me.

Reply:
Skeptical of what, exactly?
It is hard for me to address your concerns if I don't know what they are.